// python -m grpc_tools.protoc -I . --python_out=. --grpc_python_out=. optimizer.proto

syntax = "proto3";

option java_multiple_files = true;

package optimizer;

// The optimizer service definition.
service Optimizer {
  // A simple RPC.
  //
  // Get the optimization of the MPC
  rpc GetMPCOptimization (MPCOptimizationRequest) returns (Reply) {}
  rpc GetMPCSimulation (SimulationRequest) returns (SimulationReply) {}
}

message MPCOptimizationRequest {
  string building = 1;
  repeated string zones = 2;

  int64 start = 3;
  int64 end = 4;

  string window = 5;

  double lambda_val = 6;

  map<string, double> starting_temperatures = 7; // zone: temperature

  string unit = 8;

}

message SimulationRequest {
  string building = 1;
  repeated string zones = 2;

  int64 start = 3;
  int64 end = 4;

  string forecasting_horizon = 5;
  string window = 6;

  double lambda_val = 7;

  map<string, double> starting_temperatures = 8; // zone: temperature

  string unit = 9;

  int64 num_runs = 10;

}



// The response message containing the optimal actions
message Reply {
  map<string, int64> actions = 1;
}

message ActionList {
  repeated int64 actions = 1;
}

message TemperatureList {
  repeated double temperatures = 1;
}


message ActionTemperatureReply {
  map<string, ActionList> actions = 1;
  map<string, TemperatureList> temperatures = 2;
}

message SimulationReply{
  repeated ActionTemperatureReply simulation_results = 1;
}